packages/cozy-client/src/registry.js

```diff
@@ -0,0 +1,55 @@
+import 'url-search-params-polyfill'
+import termUtils from './terms'
+import { APP_TYPE } from './constants'
+
+const queryPartFromOptions = options => {
+  const query = new URLSearchParams(options).toString()
+  return query ? `?${query}` : ''
+}
+
+class Registry {
+  constructor(options) {
+    if (!options.client) {
+      throw new Error('Need to pass a client to instantiate a Registry API.')
+    }
+    this.client = options.client
+  }
+  /**
+   * Installs or updates an app from a source.
+   *
+   * Accepts the terms if the app has them.
+
+   * @param  {RegistryApp} app - App to be installed 
+   * @param  {string} source - String (ex: registry://drive/stable)
+   * @return {Promise}
+   */
+  async installApp(app, source) {
+    const { slug, type, terms } = app
+    const searchParams = {}
+    const isUpdate = app.installed
+    if (isUpdate) searchParams.PermissionsAcked = isUpdate
+    if (source) searchParams.Source = source
+    const querypart = queryPartFromOptions(searchParams)
+    if (terms) {
+      await termUtils.save(this.client, terms)
+    }
+    const verb = app.installed ? 'PUT' : 'POST'
+    const route = type === APP_TYPE.KONNECTOR ? 'konnectors' : 'apps'
```

Comment:
DRY: This probably should be an helper somewhere. It is called 2 times and has its own internal logic.